Modules/_pickle.c

```diff
@@ -4020,7 +4024,29 @@ save(PicklerObject *self, PyObject *obj, int pers_save)
         status = save_tuple(self, obj);
         goto done;
     }
-    else if (type == &PyType_Type) {
+    /*  The switch-on-type statement ends here because the next three
+     *  conditions are not exclusive anymore. If reducer_override returns
+     *  NotImplemented, then we must fallback to save_type or save_global
+     *  */
+    if (self->_reducer_override != NULL) {
+        PyObject *reduce_value = NULL;
+        reduce_value = PyObject_CallFunctionObjArgs(self->_reducer_override,
+                                                    obj, NULL);
+        if (reduce_value == NULL){
+            goto error;
+        }
+
+        if (reduce_value != Py_NotImplemented){
+            status = save_reduce(self, reduce_value, obj);
+            Py_DECREF(reduce_value);
+            if (status < 0)
```

Comment:
Curly braces should be used here (as per [PEP 7](https://www.python.org/dev/peps/pep-0007/#code-lay-out)).